src/pages/index.js

```diff
@@ -76,44 +81,159 @@ const cardSection = new Section(
 /* ------------------------------------------------------------------------------------------- */
 
 function createCard(data) {
-  const card = new Card(data, "#card__template", handleImageClick);
+  const card = new Card(
+    data,
+    "#card__template",
+    handleImageClick,
+    handleDeleteCard,
+    handleLikeCard,
+    handleRemoveLikeCard
+  );
+  console.log("Check owner of card just created", card.owner);
+  // console.log("Check created cardId", data._id);
   return card.getView();
 }
 
 function handleImageClick(data) {
   previewCardModal.open(data);
+  // console.log("Check here for id", data.id);
+}
+
+function handleRemoveLikeCard(card) {
+  api
+    .removeLikeCard(card.getId())
+    .then((response) => {
+      console.log("Api response", response);
+      card.likeStatus = response.isLiked;
+      card.setLikeState();
+      // card._handleLikeIcon();
+      // card.removeLikeCard();
+      console.log("Removed Like", card.getId());
+    })
+    .catch((err) => console.log("Error in removing a Like", err));
+}
+
+function handleLikeCard(card) {
+  api
+    .likeCard(card.getId())
+    .then((response) => {
+      console.log("Api resposnse", response);
+      card.likeStatus = response.isLiked;
+      // card._handleLikeIcon();
+      // card.likeCard();
+      card.setLikeState();
+      console.log("Liked", card.getId());
+    })
+    .catch((err) => console.log("Error in Liking a card", err));
+}
+
+function handleDeleteCard(card) {
+  popupConfirmDelete.open();
+  popupConfirmDelete.setSubmitCallback(() => {
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.handlerDeleteCard();
+        console.log("Deleted", card.getId());
+        popupConfirmDelete.close();
+      })
+      .catch((err) => console.error("Error in deleting card", err));
+  });
 }
 
+// function handleLikeCard(card){
+//   api.likeCard(card.getId())
+// }
+
 /* ------------------------------------------------------------------------------------------- */
 /*                                  Event Handlers                                             */
 /* ------------------------------------------------------------------------------------------- */
 
 function handleProfileFormSubmit(inputData) {
-  userInfo.setUserInfo({
-    profileNameData: inputData.title,
-    profileJobData: inputData.description,
-  });
+  profileEditModal.renderLoading(true);
+  api
+    .updateUserInfo({
+      name: inputData.title,
+      about: inputData.description,
+    })
+    .then((res) => {
+      console.log("Visalize res", res);
+      // console.log("Visualize res.title", res._id);
+      // console.log("Visualize res", res._id);
+      userInfo.setUserInfo({
+        // profileNameData: inputData.title,
+        profileNameData: res.name,
+        // profileJobData: inputData.description,
+        profileJobData: res.about,
+      });
+      editModalForm.reset();
+      profileEditModal.close();
+    })
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**